ocaml/xapi/xapi_vm_lifecycle.ml

```diff
@@ -496,10 +499,18 @@ let check_operation_error ~__context ~ref =
       else None) in
 
   (* If a PCI device is passed-through, check if the operation is allowed *)
-  let current_error = check current_error (fun () ->
-      if vmr.Db_actions.vM_attached_PCIs <> []
-      then check_pci ~op ~ref_str
-      else None) in
+  let current_error = check current_error @@ fun () ->
+    let sriov_pcis = nvidia_sriov_pcis ~__context vmr.Db_actions.vM_VGPUs in
+    let is_not_sriov pci = not @@ List.mem pci sriov_pcis in
+    let pcis = vmr.Db_actions.vM_attached_PCIs in
+    match op with
+    | `suspend
+    | `checkpoint
+    | `pool_migrate
+    | `migrate_send when List.exists is_not_sriov pcis ->
```

Comment:
Is the intention here for the `when` to apply to all ops?

Reply:
Yes. Is this not what is happening?

Reply:
It is, having the variants listed on separate rows made me think the intent was for it to apply just to the last variant, perhaps some parenthesis would help to make the intent explicit?

Reply:
I've made the formatting more suggestive.